# Ejercicio 5. NumPy

<b> Alumna. Daniela Lizeth Molina Barrera </b>

1. Revisar los otros archivos csv de la página del gobierno de Nueva Zelanda (https://www.stats.govt.nz/large-datasets/csv-files-for-download/) y elegir alguno que te sea interesante. Utilizar 5 funciones de NumPy que te resulten útiles para analizar el dataset, como lo realizamos en clase con el csv de COVID19 y trading. Como sugerencia: no te limites a utilizar las funciones vistas en clase, también puedes explorar otras funciones interesantes como np.concatenate () o np.dot (). La lista de funciones de NumPy está aquí: https://numpy.org/doc/stable/reference/routines.html La documentación oficial completa de NumPy la encuentras aquí: https://numpy.org/doc/stable/index.html.

In [18]:
#pip es el paquete instalador de Python
import pip

In [19]:
#Instalar Numpy usando el instalador pip
!pip install numpy

In [20]:
#Importar NumPy
import numpy as np

In [21]:
#Utilizaremos la función np.genfromtxt() para importar el archivo.
marriages= np.genfromtxt('marriages.csv',delimiter=',', skip_header= 1)
divorces = np.genfromtxt('divorces.csv',delimiter = ',', skip_header = 1)

In [22]:
#mostraremos el archivo que acabamos de importar
marriages

array([[ 2005.,    nan, 20718.],
       [ 2006.,    nan, 21798.],
       [ 2007.,    nan, 21810.],
       [ 2008.,    nan, 22275.],
       [ 2009.,    nan, 21939.],
       [ 2010.,    nan, 21213.],
       [ 2011.,    nan, 20532.],
       [ 2012.,    nan, 20823.],
       [ 2013.,    nan, 19425.],
       [ 2014.,    nan, 20175.],
       [ 2015.,    nan, 19947.],
       [ 2016.,    nan, 20235.],
       [ 2017.,    nan, 20685.],
       [ 2018.,    nan, 20949.],
       [ 2019.,    nan, 19071.],
       [ 2020.,    nan, 16779.]])

In [23]:
divorces

array([[ 2005.,    nan,  9972.],
       [ 2006.,    nan, 10065.],
       [ 2007.,    nan,  9651.],
       [ 2008.,    nan,  9720.],
       [ 2009.,    nan,  8754.],
       [ 2010.,    nan,  8901.],
       [ 2011.,    nan,  8583.],
       [ 2012.,    nan,  8826.],
       [ 2013.,    nan,  8319.],
       [ 2014.,    nan,  8232.],
       [ 2015.,    nan,  8520.],
       [ 2016.,    nan,  8169.],
       [ 2017.,    nan,  8001.],
       [ 2018.,    nan,  7455.],
       [ 2019.,    nan,  8388.],
       [ 2020.,    nan,  7707.]])

In [24]:
#convertimos los nan en 0
marriages[np.isnan(marriages)] = 0

In [25]:
marriages

array([[ 2005.,     0., 20718.],
       [ 2006.,     0., 21798.],
       [ 2007.,     0., 21810.],
       [ 2008.,     0., 22275.],
       [ 2009.,     0., 21939.],
       [ 2010.,     0., 21213.],
       [ 2011.,     0., 20532.],
       [ 2012.,     0., 20823.],
       [ 2013.,     0., 19425.],
       [ 2014.,     0., 20175.],
       [ 2015.,     0., 19947.],
       [ 2016.,     0., 20235.],
       [ 2017.,     0., 20685.],
       [ 2018.,     0., 20949.],
       [ 2019.,     0., 19071.],
       [ 2020.,     0., 16779.]])

In [26]:
# convertimos los nan en 0
divorces[np.isnan(divorces)] = 0

In [27]:
divorces

array([[ 2005.,     0.,  9972.],
       [ 2006.,     0., 10065.],
       [ 2007.,     0.,  9651.],
       [ 2008.,     0.,  9720.],
       [ 2009.,     0.,  8754.],
       [ 2010.,     0.,  8901.],
       [ 2011.,     0.,  8583.],
       [ 2012.,     0.,  8826.],
       [ 2013.,     0.,  8319.],
       [ 2014.,     0.,  8232.],
       [ 2015.,     0.,  8520.],
       [ 2016.,     0.,  8169.],
       [ 2017.,     0.,  8001.],
       [ 2018.,     0.,  7455.],
       [ 2019.,     0.,  8388.],
       [ 2020.,     0.,  7707.]])

# Algunas funciones de NumPy....

In [40]:
valores_marriages = marriages [1:, 2]
valores_marriages

array([21798., 21810., 22275., 21939., 21213., 20532., 20823., 19425.,
       20175., 19947., 20235., 20685., 20949., 19071., 16779.])

In [42]:
valores_divorces = divorces [1:, 2]
valores_divorces

array([10065.,  9651.,  9720.,  8754.,  8901.,  8583.,  8826.,  8319.,
        8232.,  8520.,  8169.,  8001.,  7455.,  8388.,  7707.])

In [46]:
# maximum
valores_marriages.max()

22275.0

*En 2008 fue el año con mas matrimonios en Nueva Zelanda con 22,275 matrimonios celebrados*

In [47]:
#minumun
valores_divorces.max()

10065.0

*En 2006 fue el año con mas divorcios en Nueva Zelanda con un total de 10,065 divorcios*

In [48]:
#min
valores_marriages.min()

16779.0

*El 2020 fue el año con menos matrimonios, con tan solo 16,779 matrimonios celebrados*

In [49]:
#min
valores_divorces.min()

7455.0

*El 2018 fue el año con la menor cantidad de divorcios,solo fueron 7,455 divorcios.*

In [50]:
#mean
valores_marriages.mean()

20510.4

*En promedio de 2005 a 2020 han sucedido 20,510 matrimonios.*

In [51]:
#mean
valores_divorces.mean()

8619.4

*En promedio de 2005 a 2020 han sucedido 8,619 divorcios.*

In [52]:
#sum
valores_marriages.sum()

307656.0

*De 2005 a 2020 en total se han celebrado 307,656 matrimonios.*

In [54]:
#sum
valores_divorces.sum()

129291.0

*De 2005 a 2020 en total se han concretado 129,291 divorcios.*